chapters/en/chapter11/4.mdx

```diff
@@ -0,0 +1,174 @@
+
+LoRA works by adding pairs of rank decomposition matrices to transformer layers, typically focusing on attention weights. During inference, these adapter weights can be merged with the base model, resulting in no additional latency overhead. LoRA is particularly useful for adapting large language models to specific tasks or domains while keeping resource requirements manageable.
+
+## Key advantages of LoRA
+
+1. **Memory Efficiency**: 
+   - Only adapter parameters are stored in GPU memory
+   - Base model weights remain frozen and can be loaded in lower precision
+   - Enables fine-tuning of large models on consumer GPUs
+
+2. **Training Features**:
+   - Native PEFT/LoRA integration with minimal setup
+   - Support for QLoRA (Quantized LoRA) for even better memory efficiency
+
+3. **Adapter Management**:
+   - Adapter weight saving during checkpoints
+   - Features to merge adapters back into base model
+
+## Loading LoRA Adapters with PEFT
+
+
+[PEFT](https://github.com/huggingface/peft) is a library that provides a unified interface for loading and managing PEFT methods, including LoRA. It allows you to easily load and switch between different PEFT methods, making it easier to experiment with different fine-tuning techniques. 
+
+Adapters can be loaded onto a pretrained model with load_adapter(), which is useful for trying out different adapters whose weights aren't merged. Set the active adapter weights with the set_adapter() function. To return the base model, you could use unload() to unload all of the LoRA modules. This makes it easy to switch between different task-specific weights.
+
+```python
+from peft import PeftModel, PeftConfig
+
+config = PeftConfig.from_pretrained("ybelkada/opt-350m-lora")
+model = AutoModelForCausalLM.from_pretrained(config.base_model_name_or_path)
+lora_model = PeftModel.from_pretrained(model, "ybelkada/opt-350m-lora")
+```
+
+<!-- TODO: Add image -->
+![lora_load_adapter](https://github.com/huggingface/smol-course/raw/main/3_parameter_efficient_finetuning/images/lora_adapter.png)
+
+## Fine-tune LLM using `trl` and the `SFTTrainer` with LoRA
+
+The [SFTTrainer](https://huggingface.co/docs/trl/sft_trainer) from `trl` provides integration with LoRA adapters through the [PEFT](https://huggingface.co/docs/peft/en/index) library. This means that we can fine-tune a model in the same way as we did with SFT, but use LoRA to reduce the number of parameters we need to train.
+
+We'll use LoRA in our example, which combines LoRA with 4-bit quantization to further reduce memory usage without sacrificing performance. The setup requires just a few configuration steps:
```

Comment:
It seems you don't do 4-bit below? If you decide to add it, I suggest mentioning it in new section above called `QLoRA` since that terminology is most common